src/async.rs

```diff
@@ -0,0 +1,169 @@
+    pub fn set_stripe_account<Str: Into<String>>(&mut self, account_id: Str) {
+        self.headers.stripe_account = Some(account_id.into());
+    }
+
+    /// Make a `GET` http request with just a path
+    pub fn get<T: DeserializeOwned + Send + 'static>(&self, path: &str) -> Response<T> {
+        let url = Client::url(path);
+        let request = self.client.get(&url).headers(self.headers());
+        send(request)
+    }
+
+    /// Make a `GET` http request with url query parameters
+    pub fn get_query<T: DeserializeOwned + Send + 'static, P: serde::Serialize>(
+        &self,
+        path: &str,
+        params: P,
+    ) -> Response<T> {
+        let url = match Client::url_with_params(path, params) {
+            Err(err) => return Box::new(future::err(err)),
+            Ok(ok) => ok,
+        };
+        let request = self.client.get(&url).headers(self.headers());
+        send(request)
+    }
+
+    /// Make a `DELETE` http request with just a path
+    pub fn delete<T: DeserializeOwned + Send + 'static>(&self, path: &str) -> Response<T> {
+        let url = Client::url(path);
+        let request = self.client.delete(&url).headers(self.headers());
+        send(request)
+    }
+
+    /// Make a `DELETE` http request with url query parameters
+    pub fn delete_query<T: DeserializeOwned + Send + 'static, P: serde::Serialize>(
+        &self,
+        path: &str,
+        params: P,
+    ) -> Response<T> {
+        let url = match Client::url_with_params(path, params) {
+            Err(err) => return Box::new(future::err(err)),
+            Ok(ok) => ok,
+        };
+        let request = self.client.delete(&url).headers(self.headers());
+        send(request)
+    }
+
+    /// Make a `POST` http request with just a path
+    pub fn post<T: DeserializeOwned + Send + 'static>(&self, path: &str) -> Response<T> {
+        let url = Client::url(path);
+        let request = self.client.post(&url).headers(self.headers());
+        send(request)
+    }
+
+    /// Make a `POST` http request with urlencoded body
+    pub fn post_form<T: DeserializeOwned + Send + 'static, F: serde::Serialize>(
+        &self,
+        path: &str,
+        form: F,
+    ) -> Response<T> {
+        let url = Client::url(path);
+        let request = self.client.post(&url).headers(self.headers());
+        let request = match with_form_urlencoded(request, &form) {
+            Err(err) => return Box::new(future::err(err)),
+            Ok(ok) => ok,
+        };
+        send(request)
+    }
+
+    fn url(path: &str) -> String {
+        format!("https://api.stripe.com/v1/{}", &path[1..])
+    }
+
+    fn url_with_params<P: serde::Serialize>(path: &str, params: P) -> Result<String, Error> {
+        let params = serde_qs::to_string(&params).map_err(Error::serialize)?;
+        Ok(format!("https://api.stripe.com/v1/{}?{}", &path[1..], params))
+    }
+
+    fn headers(&self) -> HeaderMap {
```

Comment:
Presumably we could change how this works such that it takes, modifies and returns a `RequestBuilder` rather than returning a `HeaderMap`.

Is there substantial enough benefit to library header parsing that it is worth making a new GitHub issue?

Reply:
probably not; you can look [here](https://docs.rs/reqwest/0.9.8/src/reqwest/async_impl/request.rs.html#103-124) to see what it does. Basically if header parsing fails, it stores the error in the request, so it becomes exposed on `send`. Readability/convenience mostly.